tests/src/test/scala/com/scylladb/migrator/alternator/DynamoDBInputFormatTest.scala

```diff
@@ -0,0 +1,79 @@
+package com.scylladb.migrator.alternator
+
+import com.amazonaws.services.dynamodbv2.model.{BillingMode, BillingModeSummary, ProvisionedThroughputDescription, TableDescription}
+import com.scylladb.migrator.readers.DynamoDB
+import org.apache.spark.sql.SparkSession
+
+class DynamoDBInputFormatTest extends munit.FunSuite {
+
+  val spark = SparkSession.builder().appName("test").master("local[*]").getOrCreate()
+
+  val GB: Long = 1024 * 1024 * 1024
+
+  test("no configured scanSegments in on-demand billing mode") {
+    checkPartitions(10)(tableSizeBytes = 1 * GB, tableProvisionedThroughput = None)
+  }
+
+  test("no configured scanSegments in on-demand billing mode and table size is 100 GB") {
+    checkPartitions(1024)(tableSizeBytes = 100 * GB, tableProvisionedThroughput = None)
+  }
+
+  test("no configured scanSegments in provisioned billing mode") {
+    checkPartitions(10)(tableSizeBytes = 1 * GB, tableProvisionedThroughput = Some((25, 25)))
+  }
+
+  test("scanSegments = 42") {
+    checkPartitions(42)(configuredScanSegments = Some(42))
+  }
+
+  test("scanSegements = 42 and maxMapTasks = 10") {
+    checkPartitions(10)(configuredScanSegments = Some(42), configuredMaxMapTasks = Some(10))
+  }
```

Comment:
Last, you can explicitly set both the number of `scanSegments` and `maxMapTasks`. In such a case, the  number of task mappers will be bound to `maxMapTasks`.